src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -228,86 +246,57 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
   }
 
   private List<Event> getEventListToRetry(List<Event> events, Response response) {
-    List<Event> eventsToRetry = events;
+    List<Event> eventsToRetry = new ArrayList<>();
     List<Event> eventsToDrop = new ArrayList<>();
     // Filter invalid event out based on the response code.
     if (response.status == Status.SUCCESS) {
-      return new ArrayList<>();
+        triggerEventCallbacks(events, response.code, "Events sent success.");
```

Comment:
I don't think we will hit on this case actually.

Reply:
I think you are right. I left it unchanged because it's there before and doesn't make things wrong. I'll remove it.